src/main/java/org/wildfly/security/auth/server/event/SecurityAuthenticationEvent.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.server.event;
+
+import org.wildfly.security.auth.server.SecurityIdentity;
+
+/**
+ * A security authentication event.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+public abstract class SecurityAuthenticationEvent extends SecurityEvent {
+
+    SecurityAuthenticationEvent(SecurityIdentity securityIdentity) {
+        super(securityIdentity);
+    }
+
+    /**
+     * Determine if the authentication was successful.
+     *
+     * @return {@code true} if the authentication was successful, {@code false} otherwise.
+     */
+    public abstract boolean isSuccessful();
+
+    /**
+     * Determine if the authentication failed.
+     *
+     * @return {@code true} if the authentication failed, {@code false} otherwise.
+     */
+    public final boolean isFailed() {
```

Comment:
I started following the RealmEvent pattern but I am thinking anyway it would be better if as a child of SecurityEvent we have SecurityDefiniteOutcomeEvent with just isSuccessful - I think that will make filtering easier.